### PR TITLE
Custom ssh encryption config

### PR DESCRIPTION
**Pull request by dgouju**
#### What is the purpose of this change?

Adding advanced configuration options for SFTP, to set Ciphers, Key Exchange and MACs algorithms.

#### Was the change discussed in an issue or in the forum before?

https://github.com/rclone/rclone/issues/6503

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [ ] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/sftp/sftp.go [hunk 1]
```diff
@@ -58,6 +58,15 @@ func init() {
 		Name:        "sftp",
 		Description: "SSH/SFTP",
 		NewFs:       NewFs,
+		Config: func(ctx context.Context, name string, m configmap.Mapper, config fs.ConfigIn) (*fs.ConfigOut, error) {
```
Comment (ncw) [inline backend/sftp/sftp.go:61]: Can you do this in NewFs looking at the decoded Options?

The config callback is for oauth.

File: backend/sftp/sftp.go [hunk 2]
```diff
@@ -327,7 +339,47 @@ and pass variables with spaces in in quotes, eg
...
+
+This must not be set if use_insecure_cipher is true.
+
+Example:
+
+    aes128-ctr aes192-ctr aes256-ctr aes128-gcm@openssh.com aes256-gcm@openssh.com
+`,
+                        Advanced: true,
+                },{
+                        Name:    "user_defined_key_exchange",
```
Comment (ncw) [inline backend/sftp/sftp.go:357]: Can you remove the user_defined_ prefixes here. They will make the config options very long and they find add anything I think.

**Review by ncw — COMMENTED** [review]
Looks very nice thank you :-)

See inline for some fixes.

I'll run the ci now

**Comment by dgouju** [comment]
Thanks, my first code in Go ;-) 
I modified it according to your comments.

**Comment by dgouju** [comment]
Should pass lint now =D

**Comment by dgouju** [comment]
@ncw is something else expected for me? In which release / when do you think it will be merged?
Thanks!

**Review by ncw — APPROVED** [review]
This looks great now - thank you.

I'll merge it now for 1.61

Sorry about the delay the 1.60 release got in the way.

**Pull request merged**